scripts/sbatch.py

```diff
@@ -0,0 +1,288 @@
+    else:  # not None, convert str to int
+        slurm_header_dict[SlurmHeader.ntasks.value] = int(
+            slurm_header_dict[SlurmHeader.ntasks.value]
+        )
+
+    if not slurm_header_dict[SlurmHeader.cpus_per_task.value]:
+        slurm_header_dict[SlurmHeader.cpus_per_task.value] = DEFAULT_CPUS_PER_TASK
+    else:
+        slurm_header_dict[SlurmHeader.cpus_per_task.value] = int(
+            slurm_header_dict[SlurmHeader.cpus_per_task.value]
+        )
+
+    if hpc == HPC.mahuika.value:
+        if not slurm_header_dict[SlurmHeader.mem_per_cpu.value]:
+            slurm_header_dict[SlurmHeader.mem_per_cpu.value] = get_default_mem_per_cpu(
+                slurm_conf
+            )
+        else:
+            slurm_header_dict[SlurmHeader.mem_per_cpu.value] = process_mem_per_cpu(
+                slurm_header_dict[SlurmHeader.mem_per_cpu.value]
+            )
+
+    return slurm_header_dict
+
+
+def get_hpc_conf(partition):
+    """Get hpc name and path to slurm.conf according to partition name in the slurm file header"""
+    if partition in [p.value for p in MauiPartition]:
+        slurm_conf = MAUI_SLUM_CONF  # dummy, not used
+        hpc = HPC.maui.value
+    elif partition in [p.value for p in MahuikaPartition]:
+        slurm_conf = MAHUIKA_SLURM_CONF
+        hpc = HPC.mahuika.value
+    else:
+        sys.exit("{} partition does not exit")
+    return hpc, slurm_conf
+
+
+def main():
+    parser = argparse.ArgumentParser()
+    parser.add_argument("slurm_to_sbatch", help="path to slurm file to be sbatched")
+    parser.add_argument(
+        "--core_hour_json",
+        default="core_hours.json",
+        help="path to core hour json file getting from dashboard.db, default is core_hours.json",
+    )
+    parser.add_argument(
+        "--user", default=getpass.getuser(), help="the user who's sbatching the job"
+    )
+    args = parser.parse_args()
+
+    assert os.path.exists(args.slurm_to_sbatch)
+    assert os.path.exists(args.core_hour_json)
+
+    # Get raw headers
+    header_dict = process_slurm_header(args.slurm_to_sbatch)
+
+    # Get hpc_name and path to slurm.conf
+    hpc, slurm_conf = get_hpc_conf(header_dict[SlurmHeader.partition.value])
+
+    # Process headers for calculation
+    header_dict = process_header_values(header_dict, slurm_conf, hpc)
+
+    # Get available hours
+    avai_hours = get_available_chours(
+        args.core_hour_json, header_dict[SlurmHeader.account.value], args.user
+    )
+
+    # Calculate requested hours
+    if hpc == HPC.mahuika.value:
+        cpu_weights, mem_weights = read_slurm_conf(
+            slurm_conf, header_dict[SlurmHeader.partition.value]
+        )
+        req_hours = calculate_requested_chours(
+            cpu_weights,
+            mem_weights,
+            header_dict[SlurmHeader.mem_per_cpu.value],
+            header_dict[SlurmHeader.ntasks.value],
+            header_dict[SlurmHeader.cpus_per_task.value],
+        )
+    elif hpc == HPC.maui.value:
```

Comment:
this if/elif statement can be removed after other functions has been updated to handle both maui/mahuika.